wire/msgtx.go

```diff
@@ -519,6 +659,23 @@ func (msg *MsgTx) BtcEncode(w io.Writer, pver uint32, enc MessageEncoding) error
 		return err
 	}
 
+	// If the encoding version is set to WitnessEncoding, and the Flags
+	// field for the MsgTx aren't 0x00, then this indicates the transaction
+	// is to be encoded using the new witness inclusionary structure
+	// defined in BIP0144.
+	if enc == WitnessEncoding && msg.HasWitness() {
+		// After the txn's Version field, we include two additional
+		// bytes specific to the witness encoding. The first byte is an
+		// always 0x00 marker byte, which allows decoders to
+		// distinguish a serialized transaction with witnesses from a
+		// regular (legacy) one. The second byte is the Flag field,
+		// which at the moment is always 0x01, but way be extended in
+		// the future to accommodate auxiliary non-committed fields.
+		if _, err := w.Write([]byte{0x00, 0x01}); err != nil {
```

Comment:
Good point, lifted into a package level variable to avoid the unnecessary overheard.